# Project AirBnB :

## From Poor Ratings to Exceptional Stays: A Data-Driven Analysis of Factors Influencing Airbnb Guest Satisfaction.

#### Project Goal

###### To identify the characteristics of low rated Airbnb listings and determine which property features are associated with higher guest satisfaction.

###### To analyze Airbnb listing characteristics that influence guest ratings and identify practical strategies for improving underperforming listings.


##### Business Question

###### What factors contribute to lower Airbnb guest ratings, and what improvements can hosts implement to increase guest satisfaction and listing performance?

##### Data set: "joyshil0599/airbnb-listing-data-for-data-science" (Kagglehub)

#### Purpose of this Notebook:
##### This notebook is loaded with the raw datasets, in which we explore their structure, and perform essential cleaning steps to prepare the data for reliable analysis. Here, we harmonize column names across the Airbnb file, handle missing or inconsistent values, remove duplicates, and create standardized fields such as listing categories, numeric price values, and review metrics. By the end of this notebook, we produce a clean, unified dataset that is ready for further exploratory analysis.

In [6]:
import os
os.listdir()


['functions.py',
 '.DS_Store',
 '02_load_and_clean_data.ipynb',
 'load_and_clean_data.ipynb',
 'load_clean_data_nadiya.ipynb',
 'explore_clean_data_username.ipynb',
 'load and clean data.ipynb',
 'load_and_clean_data_username.ipynb',
 '.ipynb_checkpoints']

In [9]:
import pandas as pd

df = pd.read_csv("../data/raw/airnb.csv")

print("Rows and columns:", df.shape)
display(df.head())


Rows and columns: (953, 7)


,Title,Detail,Date,Price(in dollar),Offer price(in dollar),Review and rating,Number of bed
0,"Chalet in Skykomish, Washington, US",Sky Haus - A-Frame Cabin,Jun 11 - 16,306.00,229.00,4.85 (531),4 beds
1,"Cabin in Hancock, New York, US",The Catskill A-Frame - Mid-Century Modern Cabin,Jun 6 - 11,485.00,170.00,4.77 (146),4 beds
2,"Cabin in West Farmington, Ohio, US",The Triangle: A-Frame Cabin for your city retreat,Jul 9 - 14,119.00,522.00,4.91 (515),4 beds
3,"Home in Blue Ridge, Georgia, US",*Summer Sizzle* 5 Min to Blue Ridge* Pets* Hot...,Jun 11 - 16,192.00,348.00,4.94 (88),5 beds
4,"Treehouse in Grandview, Texas, US",Luxury Treehouse Couples Getaway w/ Peaceful V...,Jun 4 - 9,232.00,196.00,4.99 (222),1 queen bed


#### Data inspection: 7 columns and 953 Rows
Title | 
Detail |
Date |
Price | 
Offer Price | 
Review and Rating |
Number of bed |
Listing-level (953 listings) | 
Mixed columns (Review and Ratings) | 
Missing values ( Offer price 787; Review and Rating 6) | 
Duplicated found 34 | dtype = String | 

In [14]:
df.columns

Index(['Title', 'Detail', 'Date', 'Price(in dollar)', 'Offer price(in dollar)',
       'Review and rating', 'Number of bed'],
      dtype='str')

In [15]:
df.isna().sum()

Title                       0
Detail                      0
Date                        0
Price(in dollar)            0
Offer price(in dollar)    787
Review and rating           6
Number of bed               0
dtype: int64

In [16]:
df.duplicated().sum()

np.int64(34)

In [17]:
df.dtypes

Title                     str
Detail                    str
Date                      str
Price(in dollar)          str
Offer price(in dollar)    str
Review and rating         str
Number of bed             str
dtype: object

##### Checking "Review and Rating" : 

In [18]:
df["Review and rating"].head(10)

0    4.85 (531)
1    4.77 (146)
2    4.91 (515)
3     4.94 (88)
4    4.99 (222)
5    4.84 (555)
6     4.64 (90)
7      5.0 (48)
8     4.84 (58)
9    4.92 (744)
Name: Review and rating, dtype: str

##### Checking "Price" column

In [19]:
df["Price(in dollar)"].unique()[:10]

<StringArray>
['306.00', '485.00', '119.00', '192.00', '232.00', '261.00', '607.00',
 '381.00',  '29.00', '154.00']
Length: 10, dtype: str

##### Checking "Number of bed"

In [20]:
df["Number of bed"].describe()


count        953
unique        33
top       2 beds
freq         221
Name: Number of bed, dtype: object

##### Check "Detail" text 

In [21]:
df["Detail"].head()

0                             Sky Haus - A-Frame Cabin
1      The Catskill A-Frame - Mid-Century Modern Cabin
2    The Triangle: A-Frame Cabin for your city retreat
3    *Summer Sizzle* 5 Min to Blue Ridge* Pets* Hot...
4    Luxury Treehouse Couples Getaway w/ Peaceful V...
Name: Detail, dtype: str

##### Check "Date" format

In [22]:
df["Date"].head()


0    Jun 11 - 16
1     Jun 6 - 11
2     Jul 9 - 14
3    Jun 11 - 16
4      Jun 4 - 9
Name: Date, dtype: str

#### CLEANING PLAN:

In [23]:
import pandas as pd
import numpy as np
import re

# Load the dataset
df = pd.read_csv("../data/raw/airnb.csv")

# Keeping a copy
df_raw = df.copy()

# Inspect dimensions and first rows
print(df.shape)
display(df.head())


(953, 7)


,Title,Detail,Date,Price(in dollar),Offer price(in dollar),Review and rating,Number of bed
0,"Chalet in Skykomish, Washington, US",Sky Haus - A-Frame Cabin,Jun 11 - 16,306.00,229.00,4.85 (531),4 beds
1,"Cabin in Hancock, New York, US",The Catskill A-Frame - Mid-Century Modern Cabin,Jun 6 - 11,485.00,170.00,4.77 (146),4 beds
2,"Cabin in West Farmington, Ohio, US",The Triangle: A-Frame Cabin for your city retreat,Jul 9 - 14,119.00,522.00,4.91 (515),4 beds
3,"Home in Blue Ridge, Georgia, US",*Summer Sizzle* 5 Min to Blue Ridge* Pets* Hot...,Jun 11 - 16,192.00,348.00,4.94 (88),5 beds
4,"Treehouse in Grandview, Texas, US",Luxury Treehouse Couples Getaway w/ Peaceful V...,Jun 4 - 9,232.00,196.00,4.99 (222),1 queen bed


#### Standardize column names

In [24]:
df.columns = (
    df.columns
      .str.strip()
      .str.lower()
      .str.replace(" ", "_")
      .str.replace(r"[^\w]+", "_", regex=True)
      .str.strip("_")
)

print(df.columns.tolist())


['title', 'detail', 'date', 'price_in_dollar', 'offer_price_in_dollar', 'review_and_rating', 'number_of_bed']


#### Check the dataset structure

In [25]:
print("Rows and columns:", df.shape)
print("\nData types:")
display(df.dtypes)

print("\nMissing values:")
display(df.isna().sum())

print("\nDuplicated rows:", df.duplicated().sum())


Rows and columns: (953, 7)

Data types:


title                    str
detail                   str
date                     str
price_in_dollar          str
offer_price_in_dollar    str
review_and_rating        str
number_of_bed            str
dtype: object


Missing values:


title                      0
detail                     0
date                       0
price_in_dollar            0
offer_price_in_dollar    787
review_and_rating          6
number_of_bed              0
dtype: int64


Duplicated rows: 34


#### Removing irrelevant columns "offer_price" and "number of bed"

In [26]:
columns_to_drop = ["offer_price", "number_of_bed"]

df = df.drop(columns=columns_to_drop, errors="ignore")

display(df.head())
print(df.columns.tolist())


,title,detail,date,price_in_dollar,offer_price_in_dollar,review_and_rating
0,"Chalet in Skykomish, Washington, US",Sky Haus - A-Frame Cabin,Jun 11 - 16,306.00,229.00,4.85 (531)
1,"Cabin in Hancock, New York, US",The Catskill A-Frame - Mid-Century Modern Cabin,Jun 6 - 11,485.00,170.00,4.77 (146)
2,"Cabin in West Farmington, Ohio, US",The Triangle: A-Frame Cabin for your city retreat,Jul 9 - 14,119.00,522.00,4.91 (515)
3,"Home in Blue Ridge, Georgia, US",*Summer Sizzle* 5 Min to Blue Ridge* Pets* Hot...,Jun 11 - 16,192.00,348.00,4.94 (88)
4,"Treehouse in Grandview, Texas, US",Luxury Treehouse Couples Getaway w/ Peaceful V...,Jun 4 - 9,232.00,196.00,4.99 (222)


['title', 'detail', 'date', 'price_in_dollar', 'offer_price_in_dollar', 'review_and_rating']


#### Clean missing values and text formatting

In [27]:
# Replace empty strings and whitespace-only values with NaN
df = df.replace(r"^\s*$", np.nan, regex=True)

# Clean all remaining text columns
text_columns = df.select_dtypes(include="object").columns

for column in text_columns:
    df[column] = df[column].str.strip()


/var/folders/t1/g1ptmxln4nx5k_rz9978lf340000gn/T/ipykernel_7238/1101536658.py:5: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  text_columns = df.select_dtypes(include="object").columns


In [28]:
display(df.isna().sum())


title                      0
detail                     0
date                       0
price_in_dollar            0
offer_price_in_dollar    787
review_and_rating          6
dtype: int64

#### Removing duplicate

In [29]:
duplicate_rows = df[df.duplicated(keep=False)].sort_values(by="title")

display(duplicate_rows.head(10))
print("Rows involved in duplicates:", len(duplicate_rows))


,title,detail,date,price_in_dollar,offer_price_in_dollar,review_and_rating
380,"Apartment in Areti, Greece",Echoes Exclusive Suite,Jun 29 - Jul 6,526.00,NaN,4.98 (40)
369,"Apartment in Areti, Greece",Echoes Exclusive Suite,Jun 29 - Jul 6,526.00,NaN,4.98 (40)
659,"Apartment in Kuala Lumpur, Malaysia",EXPRESSIONZ-1Bedroom-Junior-Suite-KLCC-Sky-Pool,Jun 1 - 6,53.00,NaN,4.78 (408)
647,"Apartment in Kuala Lumpur, Malaysia",EXPRESSIONZ-1Bedroom-Junior-Suite-KLCC-Sky-Pool,Jun 1 - 6,53.00,NaN,4.78 (408)
171,"Apartment in Marina del Rey, California, US",Romantic Ocean Penthouse w/ panoramic marina v...,Jun 1 - 8,247.00,NaN,4.77 (111)
188,"Apartment in Marina del Rey, California, US",Romantic Ocean Penthouse w/ panoramic marina v...,Jun 1 - 8,247.00,NaN,4.77 (111)
584,"Apartment in PHUKET, Thailand",★Studio in Seafront Private Villa w/infinity p...,Jun 30 - Jul 5,53.00,NaN,4.83 (93)
599,"Apartment in PHUKET, Thailand",★Studio in Seafront Private Villa w/infinity p...,Jun 30 - Jul 5,53.00,NaN,4.83 (93)
683,"Apartment in Quận 4, Vietnam",DANNY’s HIGH VIEW STUDIO near District1 Ben Th...,Jun 3 - 8,44.00,NaN,4.78 (167)
665,"Apartment in Quận 4, Vietnam",DANNY’s HIGH VIEW STUDIO near District1 Ben Th...,Jun 3 - 8,44.00,NaN,4.78 (167)


Rows involved in duplicates: 119


In [31]:
#df = df.drop_duplicates(keep="first").reset_index(drop=True)

print("Shape after removing duplicates:", df.shape)
print("Remaining duplicated rows:", df.duplicated().sum())


Shape after removing duplicates: (892, 6)
Remaining duplicated rows: 0


#### Split  review_and_rating 

In [32]:
review_rating_text = df["review_and_rating"].fillna("").str.lower()


In [33]:
df["rating"] = pd.to_numeric(
    review_rating_text.str.extract(
        r"(?P<rating>\d+(?:[.,]\d+)?)\s*(?:stars?|rating)?",
        expand=False
    ).str.replace(",", ".", regex=False),
    errors="coerce"
)


In [34]:
df["review"] = pd.to_numeric(
    review_rating_text.str.extract(
        r"(?P<review>\d[\d,]*)\s*(?:reviews?|review)",
        expand=False
    ).str.replace(",", "", regex=False),
    errors="coerce"
)


In [35]:
display(df[["review_and_rating", "rating", "review"]].head(10))


,review_and_rating,rating,review
0,4.85 (531),4.85,NaN
1,4.77 (146),4.77,NaN
2,4.91 (515),4.91,NaN
3,4.94 (88),4.94,NaN
4,4.99 (222),4.99,NaN
5,4.84 (555),4.84,NaN
6,4.64 (90),4.64,NaN
7,5.0 (48),5.00,NaN
8,4.84 (58),4.84,NaN
9,4.92 (744),4.92,NaN


In [36]:
print(df["rating"].describe())

invalid_ratings = df[
    (df["rating"].notna()) &
    ((df["rating"] < 0) | (df["rating"] > 5))
]

display(invalid_ratings)


count    871.000000
mean       4.863203
std        0.137434
min        3.670000
25%        4.820000
50%        4.890000
75%        4.960000
max        5.000000
Name: rating, dtype: float64


,title,detail,date,price_in_dollar,offer_price_in_dollar,review_and_rating,rating,review


#### Split  date  into day and month

In [37]:
display(df["date"].dropna().head(10))


0       Jun 11 - 16
1        Jun 6 - 11
2        Jul 9 - 14
3       Jun 11 - 16
4         Jun 4 - 9
5       Jun 21 - 26
6         Jun 4 - 9
7        Jun 8 - 15
8         Jul 3 - 8
9    Jun 26 - Jul 1
Name: date, dtype: str

In [38]:
parsed_date = pd.to_datetime(df["date"], errors="coerce")

df["day"] = parsed_date.dt.day
df["month"] = parsed_date.dt.month


/var/folders/t1/g1ptmxln4nx5k_rz9978lf340000gn/T/ipykernel_7238/2222532880.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  parsed_date = pd.to_datetime(df["date"], errors="coerce")


In [39]:
date_parts = df["date"].str.extract(
    r"(?i)(?P<month_name>[A-Za-z]+)\s+(?P<day>\d{1,2})"
)

df["day"] = pd.to_numeric(date_parts["day"], errors="coerce")
df["month_name"] = date_parts["month_name"].str.title()


In [40]:
print("Unparsed dates:", parsed_date.isna().sum())
display(df.loc[parsed_date.isna(), ["date"]].head())


Unparsed dates: 59


,date
9,Jun 26 - Jul 1
20,Aug 31 - Sep 5
57,Jun 27 - Jul 4
72,Aug 27 - Sep 1
73,Jun 25 - Jul 1


#### Convert  price  into a numeric column

In [44]:
df["price_clean"] = (
    df["price_in_dollar"]
      .astype("string")
      .str.replace(r"[^\d,.\-]", "", regex=True)
      .str.replace(",", "", regex=False)
)

df["price_in_dollar"] = pd.to_numeric(df["price_clean"], errors="coerce")

df = df.drop(columns="price_clean")

In [45]:
display(df[["price_in_dollar"]].head())
print(df["price_in_dollar"].describe())
print("Missing prices:", df["price_in_dollar"].isna().sum())


,price_in_dollar
0,306.0
1,485.0
2,119.0
3,192.0
4,232.0


count         892.0
mean     172.368834
std      142.793359
min            16.0
25%           85.75
50%           135.0
75%           221.0
max          1463.0
Name: price_in_dollar, dtype: Float64
Missing prices: 0


#### Split  title  into property type and city

In [46]:
title_parts = df["title"].str.extract(
    r"(?i)^(?P<property_type>.+?)\s+in\s+(?P<city>.+?)\s*$"
)

df["property_type"] = title_parts["property_type"].str.strip()
df["city"] = title_parts["city"].str.strip()
df["country"] = "US"


In [47]:
display(
    df[["title", "property_type", "city", "country"]].head(10)
)


,title,property_type,city,country
0,"Chalet in Skykomish, Washington, US",Chalet,"Skykomish, Washington, US",US
1,"Cabin in Hancock, New York, US",Cabin,"Hancock, New York, US",US
2,"Cabin in West Farmington, Ohio, US",Cabin,"West Farmington, Ohio, US",US
3,"Home in Blue Ridge, Georgia, US",Home,"Blue Ridge, Georgia, US",US
4,"Treehouse in Grandview, Texas, US",Treehouse,"Grandview, Texas, US",US
5,"Tiny home in Puerto Escondido, Mexico",Tiny home,"Puerto Escondido, Mexico",US
6,"Home in Saco, Maine, US",Home,"Saco, Maine, US",US
7,"Chalet in Grand Étang, Canada",Chalet,"Grand Étang, Canada",US
8,"Room in Mexico City, Mexico",Room,"Mexico City, Mexico",US
9,"Tiny home in Sandy Valley, Nevada, US",Tiny home,"Sandy Valley, Nevada, US",US


In [48]:
#Handling titles with different patterns
title_not_parsed = df[
    df["property_type"].isna() | df["city"].isna()
]

display(title_not_parsed[["title"]].head(20))
print("Titles not parsed:", len(title_not_parsed))


,title


Titles not parsed: 0


In [49]:
title_parts = df["title"].str.extract(
    r"(?i)^(?P<property_type>.+?)\s+in\s+"
    r"(?P<city>[^,]+)"
    r"(?:,\s*(?P<country>[A-Za-z]{2}))?\s*$"
)

df["property_type"] = title_parts["property_type"].str.strip()
df["city"] = title_parts["city"].str.strip()
df["country"] = title_parts["country"].fillna("US").str.upper()


#### Create an amenities column from  detail 

In [51]:
amenity_dictionary = {
    "pool": "Pool",
    "jacuzzi": "Jacuzzi",
    "hot tub": "Jacuzzi",
    "ocean view": "Ocean view",
    "sea view": "Ocean view",
    "ocean front": "Ocean front",
    "beachfront": "Ocean front",
    "Sauna": "sauna",
    "Nature": "nature",
    "Kayak": "kayak" , 
    "Waterfront": "waterfront" ,
    "Fireplace": "fireplace",
    "FREE Parking": "free parking" ,
    "wifi": "Wi-Fi",
    "wi-fi": "Wi-Fi",
    "kitchen": "Kitchen",
    "parking": "Parking",
    "air conditioning": "Air conditioning",
    "ac": "Air conditioning"
}


In [52]:
def extract_amenities(text):
    if pd.isna(text):
        return np.nan

    text = str(text).lower()
    found = []

    for keyword, label in amenity_dictionary.items():
        if keyword in text and label not in found:
            found.append(label)

    return ", ".join(found) if found else np.nan


In [53]:
df["amenities"] = df["detail"].apply(extract_amenities)

display(df[["detail", "amenities"]].head(10))


,detail,amenities
0,Sky Haus - A-Frame Cabin,NaN
1,The Catskill A-Frame - Mid-Century Modern Cabin,NaN
2,The Triangle: A-Frame Cabin for your city retreat,NaN
3,*Summer Sizzle* 5 Min to Blue Ridge* Pets* Hot...,Jacuzzi
4,Luxury Treehouse Couples Getaway w/ Peaceful V...,Air conditioning
5,Casa Tiny near Casa Wabi,NaN
6,The Beach House. Oceanfront & Sunrise Views!,Air conditioning
7,• Cedar Peak • 2 Bedroom Barrier-Free Chalet,NaN
8,"Room in apartment, traditional neighborhood",NaN
9,Peacock Tiny House near Las Vegas,Air conditioning


#### Remove or retain original columns

In [63]:
final_columns = [
    "property_name",
    "property_type",
    "city",
    "country",
    "detail",
    "amenities",
    "date",
    "day",
    "month",
    "price",
    "rating",
    "review"
]

# Keep only columns that actually exist
final_columns = [column for column in final_columns if column in df.columns]

airbnb_clean = df[final_columns].copy()

display(airbnb_clean.head())
print(airbnb_clean.shape)


,property_type,city,country,detail,amenities,date,day,month,rating,review
0,NaN,NaN,US,Sky Haus - A-Frame Cabin,NaN,Jun 11 - 16,11,6.0,4.85,NaN
1,NaN,NaN,US,The Catskill A-Frame - Mid-Century Modern Cabin,NaN,Jun 6 - 11,6,6.0,4.77,NaN
2,NaN,NaN,US,The Triangle: A-Frame Cabin for your city retreat,NaN,Jul 9 - 14,9,7.0,4.91,NaN
3,NaN,NaN,US,*Summer Sizzle* 5 Min to Blue Ridge* Pets* Hot...,Jacuzzi,Jun 11 - 16,11,6.0,4.94,NaN
4,NaN,NaN,US,Luxury Treehouse Couples Getaway w/ Peaceful V...,Air conditioning,Jun 4 - 9,4,6.0,4.99,NaN


(892, 10)


In [65]:
print(airbnb_clean.columns.tolist())


['property_type', 'city', 'country', 'detail', 'amenities', 'date', 'day', 'month', 'rating', 'review']


In [69]:
print("Original df columns:")
print(df_raw.columns.tolist())

print("\nCurrent df columns:")
print(df.columns.tolist())


Original df columns:
['Title', 'Detail', 'Date', 'Price(in dollar)', 'Offer price(in dollar)', 'Review and rating', 'Number of bed']

Current df columns:
['title', 'detail', 'date', 'price_in_dollar', 'offer_price_in_dollar', 'review_and_rating', 'rating', 'review', 'day', 'month', 'month_name', 'property_type', 'city', 'country', 'amenities']


In [70]:
df = df.rename(columns={
    "price_in_dollar": "price"
})


In [71]:
print(df.columns.tolist())


['title', 'detail', 'date', 'price', 'offer_price_in_dollar', 'review_and_rating', 'rating', 'review', 'day', 'month', 'month_name', 'property_type', 'city', 'country', 'amenities']


In [72]:
df["price"] = (
    df["price"]
    .astype("string")
    .str.replace(r"[^\d,.\-]", "", regex=True)
    .str.replace(",", "", regex=False)
)

df["price"] = pd.to_numeric(
    df["price"],
    errors="coerce"
)


In [73]:
print(df["price"].dtype)
display(df[["price"]].head())


Float64


,price
0,306.0
1,485.0
2,119.0
3,192.0
4,232.0


In [74]:
final_columns = [
    "title",
    "property_type",
    "city",
    "country",
    "detail",
    "amenities",
    "date",
    "day",
    "month",
    "price",
    "rating",
    "review"
]

airbnb_clean = df[final_columns].copy()

print(airbnb_clean.columns.tolist())


['title', 'property_type', 'city', 'country', 'detail', 'amenities', 'date', 'day', 'month', 'price', 'rating', 'review']


In [75]:
# Numeric decimal columns
for column in ["price", "rating", "review"]:
    airbnb_clean[column] = pd.to_numeric(
        airbnb_clean[column],
        errors="coerce"
    )

# Nullable integer columns
for column in ["day", "month"]:
    airbnb_clean[column] = pd.to_numeric(
        airbnb_clean[column],
        errors="coerce"
    ).astype("Int64")


In [76]:
print(airbnb_clean.columns.tolist())
display(airbnb_clean.head())
display(airbnb_clean.dtypes)

['title', 'property_type', 'city', 'country', 'detail', 'amenities', 'date', 'day', 'month', 'price', 'rating', 'review']


,title,property_type,city,country,detail,amenities,date,day,month,price,rating,review
0,"Chalet in Skykomish, Washington, US",NaN,NaN,US,Sky Haus - A-Frame Cabin,NaN,Jun 11 - 16,11,6,306.0,4.85,NaN
1,"Cabin in Hancock, New York, US",NaN,NaN,US,The Catskill A-Frame - Mid-Century Modern Cabin,NaN,Jun 6 - 11,6,6,485.0,4.77,NaN
2,"Cabin in West Farmington, Ohio, US",NaN,NaN,US,The Triangle: A-Frame Cabin for your city retreat,NaN,Jul 9 - 14,9,7,119.0,4.91,NaN
3,"Home in Blue Ridge, Georgia, US",NaN,NaN,US,*Summer Sizzle* 5 Min to Blue Ridge* Pets* Hot...,Jacuzzi,Jun 11 - 16,11,6,192.0,4.94,NaN
4,"Treehouse in Grandview, Texas, US",NaN,NaN,US,Luxury Treehouse Couples Getaway w/ Peaceful V...,Air conditioning,Jun 4 - 9,4,6,232.0,4.99,NaN


title                str
property_type        str
city                 str
country              str
detail               str
amenities            str
date                 str
day                Int64
month              Int64
price            Float64
rating           float64
review           float64
dtype: object

In [77]:
airbnb_clean.to_csv("airbnb_clean.csv", index=False)


In [78]:
duplicates_removed = df_raw[df_raw.duplicated(keep=False)]
duplicates_removed.to_csv("airbnb_duplicates_review.csv", index=False)
